### ISSUE TITLE
conan-libxml2 consumption

### ISSUE THREAD

**Issue by oleurodecision** [bincrafters/community#958]
### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **conan-libxml2/2.9.9**

### Question Description
[note: sorry, mac users, I can't test with this OS so won't discuss about that, please feel free to comment or update on this subject]
[note: the question is not specifically tied to conan-libxml2, but let's start with it]

Open question notably after #930 and the removal, two weeks ago, by @uilianries  of the _FindLibXml2.cmake_ previously shipped within `libxml2/2.9.9@bincrafters/stable` packages (outdated packaging approach, so +1 for the job @uilianries  ;))

now, the current situation is :
- the (official) libxml2 is just a mess of configuration and packaging stuff :
  - under linux, it can generate ("hand-written") config file for CMake (but misses config-version file, though) or for pkg-config, while the project using Autotools for its own build configuration
  - under windows, no config is generated, and the project uses some configure.js script to set up the build configuration
  - moving to some 'modern' multi-platform configuration system like CMake does not seem to be [a priority](https://gitlab.gnome.org/GNOME/libxml2/issues/24) or even really wanted ... ( even with [simpler fix for proper cmake config](https://gitlab.gnome.org/GNOME/libxml2/issues/51))
- calling `find_package(LibXml2 MODULE)` to use the _FindLibXml2.cmake_ module shipped with CMake will work with pkgconfig - only generated under linux - or by manually setting some path variables (mainly guessed from conanbuildinfo), something that should (must?) be avoided
- calling `find_package(LibXml2 CONFIG)` to use the config will only work under linux
- using cmake_find_package generator could be THE trick but ... as it generates the _Find{requirement}.cmake_ modules for all requirements, it will for instance generates a _Findboost.cmake_, that will not be used under linux when calling `find_package(Boost)`, but will fail under windows due to its case-insensitive FS

So for now, we currently tested two approaches : 
- we generate the CMake Find modules using cmake_find_package generator, then delete the unexpected ones, to allow the package to be build. The main issue with this approach is that the change has to be made for all (directly or indirectly) libxml2 consuming projects. Not cleary great, but do the stuff for now.
- we generate the CMake config file (could also be done for with pkg-config config file) within the recipe for windows builds using python. This allows us to have the same, standard behaviour under all OSes (pure CMake configuration), but it still has some hack-a-thing smell.

(note: both diffs can be published if desired, with pleasure)

Are there better ways to do this ?
What should be the adequate way of dealing with such libs ?

Anyway, we're doing the backstage work to package all our components and projects in our company, and it's clearly an enthusiastic move! We are as flexible as expected (using or not using conan in configuration), new devs can start working on projects with just a pinch of commands (`git clone`, `conan install`) under different OSes and compilers... When everything will be all set up, I hope we'll have time to document and testimony the changes (along with CMake modernization and other technical stuff). A big thanks for the work !

**Comment by uilianries** [comment]
I would say Conan generators are the best option. If you can't use **conan_find_package** because the name, maybe you can use **conan_cmake_paths**. Or, you could patch libxml2 to fix that FindLibXml2 provided by them and add again to the conan package.

Also, there is an effort to provide custom names for cmake find package files:

https://github.com/conan-io/conan/pull/5598
https://github.com/conan-io/conan/pull/5408/files

After these changes I think libxml2 and all other packages which are using CamelCase for naming will be able to work finely with Conan generators.

**Comment by Croydon** [comment]
This recipe is now in the [Conan Center Index](https://github.com/conan-io/conan-center-index/).

Please migrate to the recipe version there. 
If this is still an issue, please create a new issue in the [Conan Center Index issue tracker](https://github.com/conan-io/conan-center-index/issues).

Thanks!

**Comment by Croydon** [comment]
This recipe is now in the [Conan Center Index](https://github.com/conan-io/conan-center-index/).

Please migrate to the recipe version there. 
If this is still an issue, please create a new issue in the [Conan Center Index issue tracker](https://github.com/conan-io/conan-center-index/issues).

Thanks!

________

Some references:
https://github.com/conan-io/conan-center-index/issues/237
https://github.com/conan-io/conan-center-index/issues/209
https://github.com/conan-io/conan-center-index/issues/208

https://github.com/conan-io/conan/issues/5984
https://github.com/conan-io/conan/pull/5973
